decision-records/2024-06-04--service-to-stop-while-waiting-for-cert.md

```diff
@@ -0,0 +1,106 @@
+**Date:** 2024-06-04<br/>
+**Author:** @sed-i
+
+## Context and Problem Statement
+When an admin forms a tls relation, there is a time window in which:
+
+1. the juju model tells us we must operate with TLS enabled; however
+2. we haven't received the server cert yet.
+
+As a result we had some code ordering issues (e.g. #414):
+- Should we update the config file with the tls config, which includes path to
+  the cert, before we got the cert, only to perfectly reflect the juju model?
+- Should we wait with the tls config and allow the workload to continue to
+  operate in plain http, in contradiction to the juju model?
+- How should we determine within the charm if "TLS is enabled"?
```

Comment:
I think the issue is one of semantics more than anything: what does it mean for the charm to be tls-enabled? 
at what point in time does, precisely, the charm have all the pieces of data it needs (in the right place) for it to start talking tls with the other end? And in those intermediate states when some of the data isn't quite there yet, but the charm is already aware of the 'desired state' and knows that we're going towards it, what should the charm do?